pantab/tests/test_reader.py

```diff
@@ -15,6 +16,19 @@ def test_read_doesnt_modify_existing_file(df, tmp_hyper):
     assert last_modified == tmp_hyper.stat().st_mtime
 
 
+def test_reports_unsupported_type():
+    """
+    Test that we report an error if we encounter an unsupported column type.
+    Previously, we did not do so but instead assumed that all unsupported columns
+    would be string columns. This led to very fascinating failures.
+    """
+    db_path = pathlib.PurePath(__file__).parent / "geography.hyper"
+    with pytest.raises(
+        Exception, match=r"Column \"x\" has unsupported datatype GEOGRAPHY"
```

Comment:
This is more of a comment but would be nice to support geography in the near term; there is the geopandas library that can be leveraged

Totally find as is though...we'll cross that bridge in the future

Reply:
I had a hard time finding a good data type for this case, as I can imagine that pantab should support all data types in the long run